### PR TITLE
Modify homepage with search bar and course options

### PR DESCRIPTION
**Pull request by anhkha2003**
- Modify homepage by adding search bar and 6 course options with images
- Allow users to redirect to list of tutors by clicking the respective major in homepage
- Edit route.ts to allow redirecting to pages with GET parameters
- Fix tutor profile schedule bug by replacing useSession with useUser
- Add major parameter to URL of tutor filter to search tutors with major by URL
- Edit the .gitignore of frontend
- Add the search functionality for Homepage (same logic with search bar in tutors page)

### CODE REVIEW

File: frontend/src/app/(ui)/dashboard/tutors/tutor-profile-schedule.tsx
Comment (fornerobj) [inline frontend/src/app/(ui)/dashboard/tutors/tutor-profile-schedule.tsx:1]: Correct usage for useUser looks like this. No longer need .data?.user? etc
Feel free to check my implementation on other pages such as the dashboard

const { user } = useUser();
const userEmail = user?.email;

File: frontend/src/app/page.tsx
Comment (fornerobj) [inline frontend/src/app/page.tsx:1]: This is a good change. Thank you for fixing my sloppy implementation

**Review by fornerobj — CHANGES_REQUESTED** [review]
tutor-profile-schedule.tsx page useUser() method needs to be fixed as described in the comment

File: frontend/src/app/(ui)/dashboard/tutors/tutor-profile-schedule.tsx
Comment (anhkha2003) [reply frontend/src/app/(ui)/dashboard/tutors/tutor-profile-schedule.tsx:1]: image
This is my change. Should it be like this?

**Pull request merged**